x-pack/platform/plugins/shared/integration_assistant/public/components/create_integration/create_integration_assistant/flyout/cel_configuration/steps/cel_confirm_settings_step/cel_confirm_step.tsx

```diff
@@ -0,0 +1,165 @@
+    const [selectedOtherPath, setSelectedOtherPath] = useState<string | undefined>();
+    const [useOtherPath, setUseOtherPath] = useState<boolean>(false);
+
+    const [selectedAuth, setSelectedAuth] = useState<string | undefined>();
+    const [specifiedAuthForPath, setSpecifiedAuthForPath] = useState<string[]>([]);
+    const [invalidAuth, setInvalidAuth] = useState<boolean>(false);
+
+    useEffect(() => {
+      const path = selectedPath !== i18n.ENTER_MANUALLY ? selectedPath : selectedOtherPath;
+      if (path) {
+        const authMethods = integrationSettings?.apiSpec?.operation(path, 'get').prepareSecurity();
+        const specifiedAuth = authMethods ? Object.keys(authMethods) : [];
+        setSpecifiedAuthForPath(specifiedAuth);
+      }
+    }, [selectedPath, selectedOtherPath, integrationSettings?.apiSpec]);
+
+    const setIntegrationValues = useCallback(
+      (settings: Partial<IntegrationSettings>) =>
+        setIntegrationSettings({ ...integrationSettings, ...settings }),
+      [integrationSettings, setIntegrationSettings]
+    );
+
+    const onChangeSuggestedPath = useCallback(
+      (path: string) => {
+        setSelectedPath(path);
+        setUseOtherPath(path === i18n.ENTER_MANUALLY);
+        if (path !== i18n.ENTER_MANUALLY) {
+          setSelectedOtherPath(undefined);
+          setIntegrationValues({ celPath: path });
+        }
+      },
+      [setSelectedPath, setUseOtherPath, setIntegrationValues]
+    );
+
+    const onChangeOtherPath = useCallback(
+      (field: EuiComboBoxOptionOption[]) => {
+        const path = field && field.length ? field[0].label : undefined;
+        setSelectedOtherPath(path);
+        setIntegrationValues({ celPath: path });
+      },
+      [setSelectedOtherPath, setIntegrationValues]
+    );
+
+    const onChangeAuth = useCallback(
+      (field: EuiComboBoxOptionOption[]) => {
+        const auth = field && field.length ? field[0].label : undefined;
+        setSelectedAuth(auth);
+
+        if (auth) {
+          const translatedAuth = translatedAuthValue(auth);
+          setIntegrationValues({ celAuth: translatedAuth.toLowerCase() as CelAuthType });
+          if (specifiedAuthForPath) {
+            setInvalidAuth(!specifiedAuthForPath.includes(translatedAuth));
+          }
+        } else {
+          setIntegrationValues({ celAuth: undefined });
+          setInvalidAuth(false);
+        }
+      },
+      [setIntegrationValues, specifiedAuthForPath]
+    );
+
+    const onGenerationCompleted = useCallback<OnComplete>(
+      (result: State['celInputResult']) => {
+        if (result) {
+          setCelInputResult(result);
+          setIsFlyoutGenerating(false);
+          setShowCelCreateFlyout(false);
+        }
+      },
+      [setCelInputResult, setIsFlyoutGenerating, setShowCelCreateFlyout]
+    );
+    const onGenerationClosed = useCallback(() => {
+      setIsFlyoutGenerating(false); // aborts generation
```

Comment:
We should also reset `celPath` and `celAuth` from `integrationSettings` when closing the modal without saving. Otherwise if the user uploads a second openAPI specs file, the initial values stored in `integrationSettings` for path and auth are still the ones selected the first time (as values are not null UI does not enforce user to select a new path)
Here, no values are selected (selected path and auth previously, then closed the modal, and upload a new one)
<img width="968" alt="Screenshot 2024-12-30 at 17 40 54" src="https://github.com/user-attachments/assets/a032ba40-fb2b-4ec4-87d4-735d13bff5d3" />
But values are initialized:
<img width="883" alt="Screenshot 2024-12-30 at 17 41 03" src="https://github.com/user-attachments/assets/2525f75b-0057-400f-b7b9-3e93b0d40c01" />